nginx-controller/controller/controller.go

```diff
@@ -661,50 +722,93 @@ func (lbc *LoadBalancerController) syncSecret(task Task) {
+				for _, ing := range nonMinions {
 					lbc.recorder.Eventf(&ing, api_v1.EventTypeWarning, "UpdatedWithError", "Configuration for %v/%v was updated, but not applied: %v", ing.Namespace, ing.Name, err)
 				}
+				for _, minion := range minions {
+					master, err := lbc.findMasterForMinion(&minion)
+					if err != nil {
+						glog.Errorf("Ignoring Ingress %v(Minion): %v", minion.Name, err)
+						continue
+					}
+					lbc.recorder.Eventf(master, api_v1.EventTypeWarning, "UpdatedWithError", "Configuration for  minion %v/%v was updated, but not applied: %v", minion.Namespace, minion.Name, err)
+					lbc.recorder.Eventf(&minion, api_v1.EventTypeWarning, "UpdatedWithError", "Configuration for %v/%v was updated, but not applied: %v", minion.Namespace, minion.Name, err)
+				}
 			} else {
 				lbc.recorder.Eventf(secret, api_v1.EventTypeNormal, "Updated", "%v was updated", key)
-				for _, ing := range ings {
+				for _, ing := range nonMinions {
 					lbc.recorder.Eventf(&ing, api_v1.EventTypeNormal, "Updated", "Configuration for %v/%v was updated", ing.Namespace, ing.Name)
 				}
+				for _, minion := range minions {
+					master, err := lbc.findMasterForMinion(&minion)
+					if err != nil {
+						glog.Errorf("Ignoring Ingress %v(Minion): %v", minion.Name, err)
+						continue
+					}
+					lbc.recorder.Eventf(master, api_v1.EventTypeNormal, "Updated", "Configuration for minion %v/%v was updated", minion.Namespace, minion.Name)
+					lbc.recorder.Eventf(&minion, api_v1.EventTypeNormal, "Updated", "Configuration for %v/%v was updated", minion.Namespace, minion.Name)
+				}
 			}
 		}
 	}
 }
 
-func (lbc *LoadBalancerController) findIngressesForSecret(secret string) ([]extensions.Ingress, error) {
-	res := []extensions.Ingress{}
+func (lbc *LoadBalancerController) findIngressesForSecret(secret string) ([]extensions.Ingress, []extensions.Ingress, error) {
+	var nonMinions []extensions.Ingress
+	var minions []extensions.Ingress
+
 	ings, err := lbc.ingLister.List()
 	if err != nil {
-		return nil, fmt.Errorf("Couldn't get the list of Ingress resources: %v", err)
+		return nil, nil, fmt.Errorf("Couldn't get the list of Ingress resources: %v", err)
 	}
 
 items:
 	for _, ing := range ings.Items {
 		if !lbc.isNginxIngress(&ing) {
 			continue
 		}
-		if !lbc.cnf.HasIngress(&ing) {
-			continue
-		}
-		for _, tls := range ing.Spec.TLS {
-			if tls.SecretName == secret {
-				res = append(res, ing)
-				continue items
+
+		if !isMinion(&ing) {
+			if !lbc.cnf.HasIngress(&ing) {
+				continue
+			}
+			for _, tls := range ing.Spec.TLS {
+				if tls.SecretName == secret {
+					nonMinions = append(nonMinions, ing)
+					continue items
+				}
+			}
+			if lbc.nginxPlus {
+				if jwtKey, exists := ing.Annotations[nginx.JWTKeyAnnotation]; exists {
+					if jwtKey == secret {
+						nonMinions = append(nonMinions, ing)
+					}
+				}
 			}
+			continue
 		}
+
+		// we're dealing with a minion
+		// only JWT secrets are allowed in a minion
```

Comment:
only JWT secrets are allowed in a minion->JWT secrets are only allowed in a minion

Reply:
nope, JWT secrets are allowed in both minions and non-minions
however, minions can only have JWT secrets.